packages/flutter_tools/lib/runner.dart

```diff
@@ -45,6 +46,8 @@ Future<int> run(
   }
 
   return runInContext<int>(() async {
+    globals.terminal.applyFeatureFlags(featureFlags);
```

Comment:
Is this better than just looking up `globals.featureFlags.isCliAnimationEnabled`inside the `AnsiTerminal` class? It could be a lazily initialized `late final` property?

Reply:
I feel like everything we set here or in `FlutterCommandRunner.runCommand()` is a gotcha easter egg.

Reply:
I guess if I'm using global variables anyway I might as well do it as a late final, sure.

Reply:
So I started doing this but I got stuck because I can't figure out what the long-term plan is here. In the PR as posted, the long-term plan would be to pass the FeatureFlags and the AnsiTerminal to the runner and have it apply one to the other, or (better) to have the code that creates the AnsiTerminal and the FeatureFlags apply one to the other before giving it to the runner, but if the AnsiTerminal doesn't get given a FeatureFlags at all then I don't understand how we replace the call to the globals in the AnsiTerminal code.